fairlearn/linear_model/_constrained_logistic.py

```diff
@@ -0,0 +1,813 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import numbers
+import warnings
+
+import numpy as np
+import pandas as pd
+from scipy import optimize
+from joblib import Parallel
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.linear_model._logistic import (
+    _logistic_loss,
+)
+from sklearn.preprocessing import LabelEncoder, OneHotEncoder
+from sklearn.utils import (
+    check_array,
+    check_consistent_length,
+    compute_class_weight,
+)
+from sklearn.utils.fixes import delayed, _joblib_parallel_args
+from sklearn.utils.multiclass import check_classification_targets
+from sklearn.utils.optimize import _check_optimize_result
+from sklearn.utils.validation import _check_sample_weight
+
+
+# Some helper check functions
+def _check_solver(solver, penalty):
+    all_solvers = ["SLSQP"]
+    if solver not in all_solvers:
+        raise ValueError(
+            "Constrained Logistic Regression supports only solvers in %s, got %s."
+            % (all_solvers, solver)
+        )
+
+    all_penalties = ["l2", "none"]
+    if penalty not in all_penalties:
+        raise ValueError(
+            "Constrained Logistic Regression supports only penalties in %s, got %s."
+            % (all_penalties, penalty)
+        )
+
+    return solver
+
+
+def _check_multi_class(multi_class):
+    if multi_class not in ("ovr"):
+        raise ValueError("multi_class should be 'ovr'. Got %s." % multi_class)
+    return multi_class
+
+
+def _sensitive_attr_constraint_cov(model, X_train, A_train, covariance_bound):
+    """
+    Calculate the covariance bound.
+
+    The covariance bound is calculated as described in
+    `Zafar et al. (2017) <https://proceedings.mlr.press/v54/zafar17a.html>`_.
+
+    Parameters
+    ----------
+    model : numpy.ndarray
```

Comment:
I do agree that it would be nicer to pass `coef` and `intercept` separately, but I'm afraid this isn't possible. The `_sensitive_attr_constraint_cov` function uses the same `x0` argument as in the `optimize.minimize` function (around lines 280-290). Changing this `x0` parameter would mess with the logistic loss function, since that one also doesn't assume that `coef` and `intercept` are passed separately (it is imported from scikit, so I don't think we want to rewrite that).